jcl/src/java.base/share/classes/java/lang/JFRHelpers.java

```diff
@@ -38,6 +40,162 @@ final class JFRHelpers {
+			timeInNanos = time;
+			break;
+		}
+		return timeInNanos;
+	}
+
+	private static void initJFRCmdlineOptions() {
+		try {
+			Class<?> dcmdStartClass = Class.forName("jdk.jfr.internal.dcmd.DCmdStart");
+			Constructor<?> constructor = dcmdStartClass.getDeclaredConstructor();
+			constructor.setAccessible(true);
+			Object dcmdStartInstance = constructor.newInstance();
+
+			/*[IF JAVA_SPEC_VERSION == 11]*/
+			String fileName = null;
+			String settings = null;
+			Long delay = null;
+			Long duration = null;
+			Boolean disk = null;
+			Long maxAge = null;
+			Long maxSize = null;
+			Boolean dumpOnExit = null;
+			String[] jfrCMDLineOptionPairs = jfrCMDLineOption.split(",");
+			for (String pair : jfrCMDLineOptionPairs) {
+				String[] configKeyValue = pair.split("=");
+				if (2 == configKeyValue.length) {
+					String key = configKeyValue[0];
+					String value = configKeyValue[1];
+					switch (key) {
+					case "filename":
+						fileName = value;
+						break;
+					case "settings":
+						settings = value;
+						break;
+					case "delay":
+						delay = convertToNanoSeconds(value);
+						break;
+					case "duration":
+						duration = convertToNanoSeconds(value);
+						break;
+					case "maxage":
+						maxAge = convertToNanoSeconds(value);
+						break;
+					case "maxsize":
+						maxSize = convertToBytes(value);
+						break;
+					case "dumponexit":
+						dumpOnExit = Boolean.valueOf(value);
+						break;
+					case "disk":
+						disk = Boolean.valueOf(value);
+						break;
+					}
+				}
+			}
+			if (null == settings) {
+				settings = "default";
+			}
+			/* Convert the string to a String array */
+			String[] settingsArray = new String[] { settings };
+
+			Method executeMethod = dcmdStartClass.getDeclaredMethod(
+				"execute",
+				String.class, //name
+				String[].class, //settings
+				Long.class, //delay
+				Long.class, //duration
+				Boolean.class, //disk
+				String.class, //path
+				Long.class, //maxAge
+				Long.class, //maxSize
+				Boolean.class, //dumpOnExit
+				Boolean.class //pathToGcRoots
+			);
+
+			executeMethod.setAccessible(true);
+			String results = (String) executeMethod.invoke(
+				dcmdStartInstance,
+				null,
+				settingsArray,
+				delay,
+				duration,
+				disk,
+				fileName,
+				maxAge,
+				maxSize,
+				dumpOnExit,
+				null
+			);
+			if (null != results) {
+				logJFR(results, 0, 2);
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION == 11 */
+
+			/*[IF JAVA_SPEC_VERSION >= 17]*/
+			Method executeMethod = dcmdStartClass.getSuperclass().getDeclaredMethod("execute", String.class, String.class, char.class);
+			executeMethod.setAccessible(true);
+
+			String[] results = (String []) executeMethod.invoke(dcmdStartInstance, "internal", jfrCMDLineOption, ',');
+			if (null != results) {
+				for (String result : results) {
+					logJFR(result, 0, 2);
+				}
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION >= 17 */
+		} catch (InvocationTargetException e) {
+			Throwable cause = e.getCause();
```

Comment:
please do seomthing like:
```
           Throwable cause = e.getCause();
			e.printStackTrace();
			throw new InternalError(cause);
```

If the failure occurs early in initialization logging will not be initialized and the error will be hidden